src/code/z_fireobj.c

```diff
@@ -115,16 +115,16 @@ void FireObj_UpdateStateTransitions(GlobalContext* globalCtx, FireObj* fire) {
     }
     if ((fire->flags & 1) && (fire->state != FIRE_STATE_3) &&
         WaterBox_GetSurface1_2(globalCtx, &globalCtx->colCtx, fire->position.x, fire->position.z, &waterY, &waterBox) &&
-        ((fire->yScale * ((void)0, 6500.0f)) < (waterY - fire->position.y))) { // Fake but IDK what else
+        (waterY - fire->position.y) > (6500.0f * fire->yScale)) {
```

Comment:
```suggestion
        (waterY - fire->position.y > 6500.0f * fire->yScale) {
```
Since everything else has brackets like this.